core/src/main/java/com/scalar/db/transaction/consensuscommit/CrudHandler.java

```diff
@@ -248,6 +256,20 @@ public void closeScanners() throws CrudException {
     }
   }
 
+  private void putIntoReadSetInSnapshot(Snapshot.Key key, Optional<TransactionResult> result) {
```

Comment:
As an optimization, in read-only mode, we don't need to add the result to the read set, since the read set is only used for write operations.